core/persist/PersistenceProvider.ts

```diff
@@ -215,3 +220,27 @@ export abstract class PersistenceProvider<S> {
         return null;
     }
 }
+
```

Comment:
Realize that most of our interfaces are at top (including pre-existing one in this file), but this seemed like an advanced/wonky case that was more distracting up top than I wanted.  Don't feel strongly about it, but that's why I threw it in at the bottom instead.